streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobGenerator.scala

```diff
@@ -123,6 +126,12 @@ class JobGenerator(jobScheduler: JobScheduler) extends Logging {
         timedOut
       }
 
+      // generate one more bacth to make sure RDD in lastJob is checkpointed.
+      if (!jobScheduler.receiverTracker.hasUnallocatedBlocks &&
+        ssc.graph.isCheckpointMissedLastTime) {
+        Thread.sleep(ssc.graph.batchDuration.milliseconds)
```

Comment:
ok, I was wondering it will be more graceful if this thread can be woke up as soon as the batch job finished, other than sleep with a fixed length of time.